packages/notebook/src/widget.ts

```diff
@@ -1077,6 +1123,7 @@ export class StaticNotebook extends WindowedList<NotebookViewModel> {
   private _notebookModel: INotebookModel | null;
   private _renderingLayout: RenderingLayout | undefined;
   private _renderingLayoutChanged = new Signal<this, RenderingLayout>(this);
+  protected contentVisibilityObserver: IntersectionObserver | null = null;
```

Comment:
This should be private and handled in `onBeforeDetach` of the same class. Apologies, I think I mistakenly pointed you to an existing `onBeforeDetach` of `Notebook` as there was not one in `StaticNotebook`. In that case I suggest we add one (and call the parent one).

```suggestion
  private _contentVisibilityObserver: IntersectionObserver | null = null;
```